mirrord/layer/src/exec.rs

```diff
@@ -125,31 +217,26 @@ pub(crate) unsafe extern "C" fn execve_detour(
     argv: *const *const c_char,
     envp: *const *const c_char,
 ) -> c_int {
-    // Do unsafe part of path conversion here.
-    let rawish_path = (!path.is_null()).then(|| CStr::from_ptr(path));
-    let mut patched_path = CString::default();
-    let final_path = patch_if_sip(rawish_path)
-        .and_then(|s| match CString::new(s) {
-            Ok(c_string) => {
-                patched_path = c_string;
-                Success(patched_path.as_ptr())
-            }
-            Err(err) => Error(Null(err)),
-        })
-        .unwrap_or(path); // Continue even if there were errors - just run without patching.
-
-    let argv_arr = Nul::new_unchecked(argv);
+    let patched_args = patch_sip_for_new_process(path, argv);
+    FN_EXECVE(patched_args.get_path(), patched_args.get_argv(), envp)
+}
 
-    // If we intercept args, we create a new array.
-    // execve takes a null terminated array of char pointers.
-    // ptr_vec will own the vector that will be passed to execve as an array.
-    let mut ptr_vec: Vec<*const c_char> = Vec::new();
-    let final_argv = intercept_tmp_dir(argv_arr)
-        .map(|new_vec| {
-            ptr_vec = new_vec; // Make sure vector still lives when we pass the pointer to execve.
-            ptr_vec.as_ptr()
-        })
-        .unwrap_or(argv);
-
-    FN_EXECVE(final_path, final_argv, envp)
+#[hook_guard_fn]
+pub(crate) unsafe extern "C" fn posix_spawn_detour(
```

Comment:
add todo for spawnp?

Reply:
Added. If we're lucky `posix_spawnp` calls `posix_spawn` the same way `execvp` calls `execve`.